File: src/net/theharrisoncrafter/Commands/Getenderchest.java
package net.theharrisoncrafter.Commands;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Getenderchest implements CommandExecutor{
	public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){
		
		if(sender instanceof Player && sender.hasPermission("GodCraft.Getenderchest")){
			Player targetPlayer;
			String version = Bukkit.getBukkitVersion();
			
			if(version.startsWith("1")){
				version.replaceFirst("1.", "");
				version.substring(0, 1);
				if(Integer.parseInt(version.substring(0, 1)) < 3){
					sender.sendMessage("This version of minecraft is too old for that!");
					return true;
				}
			}
			
			if(args.length > 1){
				return false;
			}
			
			if(args.length < 1){
				targetPlayer = (Player) sender;
			}
			else{			
				targetPlayer = Bukkit.getPlayer(args[0]);
			}
			
			if(targetPlayer == null){
				sender.sendMessage(ChatColor.RED + "That player is not online!");
				return true;
			}
			
			((Player) sender).openInventory(targetPlayer.getEnderChest());
			return true;
		}
		else{
			sender.sendMessage(ChatColor.RED + "You can not send that command!");
			return true;
		}
	}
}

File: src/main/kotlin/co/orre/godcraft/Commands/GetEnderChest.kt
package co.orre.godcraft.Commands

import co.orre.godcraft.God
import org.bukkit.Bukkit
import org.bukkit.ChatColor as CC
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class GetEnderChest(val plugin: God) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, commandLabel: String, args: Array<String>): Boolean {

        if (sender is Player && sender.hasPermission("godcraft.get_ender_chest")) {
            if (args.size > 1) return false
            val targetPlayer: Player? = if (args.isEmpty()) sender else Bukkit.getPlayer(args[0])

            if (targetPlayer == null) {
                sender.sendMessage("${CC.RED}That player could not be found!")
                return true
            }

            sender.openInventory(targetPlayer.enderChest)
            plugin.logDebug("${sender.name} opened ${targetPlayer.name}'s Ender Chest")
            return true
        }
        sender.sendMessage("${CC.RED}You do not have permissions to do that!")
        return true
    }
}
